RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -0,0 +1,182 @@
+            _variableNamesAccessibleToProcedureContext = GetVariableNamesAccessibleToProcedureContext(_target.Context.Parent.Parent);
+        }
+
+        public override bool CanFixInModule { get { return false; } }
+        public override bool CanFixInProject { get { return false; } }
+
+        //This function exists solely to support unit testing 
+        public void TESTONLY_FixUsingAutoGeneratedName(string suggestedName = "")
+        {
+            //Prevent the popup dialog and forces the use of the Suggested Name
+            _isQuickFixUnitTest = true;
+            if (suggestedName.Length > 0)
+            {
+                _localCopyVariableName = suggestedName;  //unit test poses as user input
+            }
+
+            Fix();
+        }
+
+        public override void Fix()
+        {
+            SetLocalCopyVariableName();
+
+            if (!ProposedLocalVariableNameIsValid() || IsCancelled)
+            {
+                return;
+            }
+
+            ReplaceAssignedByValParameterReferences();
+
+            InsertLocalVariableDeclarationAndAssignment();
+        }
+
+        private void SetLocalCopyVariableName()
+        {
+            using (var view = new AssignedByValParameterQuickFixDialog(_target, Selection))
+            {
+                view.NewName = _localCopyVariableName;
+                view.IdentifierNamesAlreadyDeclared = _variableNamesAccessibleToProcedureContext;
+                if (!_isQuickFixUnitTest)
+                {
+                    view.ShowDialog();
+                    IsCancelled = view.DialogResult == DialogResult.Cancel;
+                }
+                if (!IsCancelled)
+                {
+                    _localCopyVariableName = view.NewName;
+                }
+            }
+        }
+
+        private bool ProposedLocalVariableNameIsValid()
+        {
+            var validator = new VariableNameValidator(_localCopyVariableName);
+            return validator.IsValidName()
+                && !_variableNamesAccessibleToProcedureContext
+                    .Any(name => name.ToUpper().Equals(_localCopyVariableName.ToUpper()));
+        }
+
+        private void ReplaceAssignedByValParameterReferences()
+        {
+            var module = Selection.QualifiedName.Component.CodeModule;
+            foreach (IdentifierReference identifierReference in _target.References)
+            {
+                module.ReplaceIdentifierReferenceName(identifierReference, _localCopyVariableName);
+            }
+        }
+        private void InsertLocalVariableDeclarationAndAssignment()
+        {
+            var blocks = QuickFixHelper.GetBlockStmtContextsForContext(_target.Context.Parent.Parent);
+            var firstBlockLineNumber =  blocks.FirstOrDefault().Start.Line;
+
+            var module = Selection.QualifiedName.Component.CodeModule;
+            module.InsertLines(firstBlockLineNumber++, BuildLocalCopyDeclaration());
```

Comment:
I can extract it.  Originally, I had a CodeModuleExtension function InsertLinesAfter( int lineNumber, string[] lines) but thought it might be overkill for inserting just two lines.  Would putting a function like this in the CodeModule extension class make sense in general?